tests/test-map.c

```diff
@@ -9,7 +9,23 @@ static int compare_fn_int(void *x, void *y) {
 	return 0;
 }
 
-void test_map(void) {
+bool predicate_fn_one(void *key) {
```

Comment:
add `static` specifier for the function signacture.
Like `static bool predicate_fn_one`

This will make the function definition local only to the [Translation unit](https://en.wikipedia.org/wiki/Translation_unit_(programming)) hence avoided multiple definition compiler errors.